nodes/persistence-node/src/services/persistence-service/PersistenceService.ts

```diff
@@ -135,73 +135,74 @@ export class PersistenceService {
     if(result === "yes") {
       await this.deps.persistenceStateManager.setIpfsHashInfo(ipfsHash, {
         ipfsHash,
-        isWrapper: true,
-        isPinned: false,
+        status: Status.Pinning,
         indexes,
       });
 
-      await this.pinCID(ipfsHash, retryCount, indexes);  
+      await this.pinWrapper(ipfsHash, retryCount, indexes);  
     } else if (result === "no") {
       await this.deps.persistenceStateManager.setIpfsHashInfo(ipfsHash, {
         ipfsHash,
-        isWrapper: false,
-        isPinned: false,
+        status: Status.NotAWrapper,
         indexes,
       });
     } else if (result === "timeout") {
-      await this.scheduleRetry(ipfsHash, retryCount, indexes);
+      await this.scheduleRetry(ipfsHash, retryCount, Status.CheckingIfWrapper, indexes);
     }
   }
   
   private async tryUntrackIpfsHash(info: TrackedIpfsHashInfo): Promise<void> {
-    if(!info.isWrapper) {
+    if(info.status === Status.NotAWrapper) {
```

Comment:
If the hash is not pinned we can just remove it. So it'll be:
`if(info.status !== Status.Pinned && info.status !== Status.Unpinning)`